lib/rouge/lexers/livescript.rb

```diff
@@ -0,0 +1,293 @@
+        # instance attributes
+        rule /(@)(#{id})/ do
+          groups Name::Variable::Instance, Name::Attribute
+          push :id
+        end
+        rule /([.])(#{id})/ do
+          groups Punctuation, Name::Attribute
+          push :id
+        end
+        rule /([.])(\d+)/ do
+          groups Punctuation, Num::Integer
+          push :id
+        end
+        rule /#{id}(?=\s*:[^:])/, Name::Attribute
+        rule /(&)(#{id})?/ do
+          groups Name::Builtin, Name::Attribute
+        end
+
+        rule /@/, Name::Variable::Instance
+        rule /:/, Punctuation
+
+        # keywords
+        rule /#{id}/ do |m|
+          if loop_control_keywords.include? m[0]
+            token Keyword
+            push :loop_control
+            next
+          elsif self.class.keywords.include? m[0]
+            token Keyword
+          elsif self.class.constants.include? m[0]
+            token Name::Constant
+          elsif self.class.builtins.include? m[0]
+            token Name::Builtin
+          elsif self.class.declarations.include? m[0]
+            token Keyword::Declaration
+          elsif /^[A-Z]/.match(m[0]) && /[^-][a-z]/.match(m[0])
+            token Name::Class
+          else
+            token Name::Other
+          end
+          push :id
+        end
+
+        # punctuation and brackets
+        rule /\](?=[!?.]|#{id})/, Punctuation, :id
+        rule /[{(\[;,]/, Punctuation, :slash_starts_regex
+        rule /[})\].]/, Punctuation
+
+        # literals
+        rule /#{int_number}[.]#{int}/, Num::Float
+        rule /0x[0-9A-Fa-f]+/, Num::Hex
+        rule /#{int}/, Num::Integer
+
+        # strings
+        rule /"""/, Str, :tdqs
+        rule /'''/, Str, :tsqs
+        rule /"/, Str, :dqs
+        rule /'/, Str, :sqs
+        rule /\\\S[^\s,;\])}]*/, Str
+      end
+
+      state :strings do
+        # all livescript strings are multi-line
+        rule /[^#\\'"]+/m, Str
+        mixin :code_escape
+        rule /\\./, Str::Escape
+        rule /#/, Str
+      end
+
+      state :double_strings do
+        rule /'/, Str
+        mixin :has_interpolation
+        mixin :strings
+      end
+
+      state :single_strings do
+        rule /"/, Str
+        mixin :strings
+      end
+
+      state :interpolation do
+        rule /}/, Str::Interpol, :pop!
+        mixin :root
+      end
+
+      state :has_interpolation do
+        rule /#\{/, Str::Interpol, :interpolation
+        rule /(#)(#{id})/ do
+          groups Str::Interpol, Name::Other
+        end
+      end
+
+      state :dqs do
+        rule /"/, Str, :pop!
+        mixin :double_strings
+      end
+
+      state :tdqs do
+        rule /"""/, Str, :pop!
+        rule /"/, Str
+        mixin :double_strings
+      end
+
+      state :sqs do
+        rule /'/, Str, :pop!
+        mixin :single_strings
+      end
+
+      state :tsqs do
+        rule /'''/, Str, :pop!
+        rule /'/, Str
+        mixin :single_strings
+      end
+    end
+  end
+end
```

Comment:
This lexer appears to by copy-pasted from the CoffeeScript lexer. Please use lexer subclassing instead, as per the README.